runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/walker/Rs2Walker.java

```diff
@@ -293,13 +293,14 @@ private static WalkerState processWalk(WorldPoint target, int distance) {
                 }
             }
 
-
             if (!doorOrTransportResult) {
                 if (!path.isEmpty()) {
                     var moveableTiles = Rs2Tile.getReachableTilesFromTile(path.get(path.size() - 1), Math.min(3, distance)).keySet().toArray(new WorldPoint[0]);
                     var finalTile = moveableTiles.length > 0 ? moveableTiles[Random.random(0, moveableTiles.length)] : path.get(path.size() - 1);
 
-                    if (Rs2Tile.isTileReachable(finalTile)) {
+
+                    Microbot.log("Distance check: " + (Rs2Player.getWorldLocation().distanceTo(target) > distance));
+                    if (Rs2Tile.isTileReachable(finalTile) && Rs2Player.getWorldLocation().distanceTo(target) > distance) {
```

Comment:
Prevent Rs2Walker.walkFastCanvas(finalTile) from running if the player is close enough the the destination.